File: backend/src/main/java/com/codeloom/backend/dao/problem/ProblemRepository.java
package com.codeloom.backend.dao.problem;

import com.codeloom.backend.model.Problem;
import org.springframework.data.repository.CrudRepository;

public interface ProblemRepository extends CrudRepository<Problem, Long>, ProblemRepositoryCustom {
    Problem findBySlug(String slug);
}

File: backend/src/main/kotlin/com/codeloom/backend/dao/ProblemRepository.kt
package com.codeloom.backend.dao

import com.codeloom.backend.model.Problem
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface ProblemRepository : CrudRepository<Problem, Long> {
    fun findBySlug(slug: String): Problem?
}